make setup.py not rely on an installed cython

jor- opened the pull request:
Referencing to issue #20, I tried to make `setup.py` not to rely on an installed `cython`.

The installation worked with an installed `cython`. Without `cython` installed, it installs `cython`. In my case, unfortunately, an error occurs in this process which seams to be related to the Intel compiler `icc` and not to `scikit-sparse` or `cython`.
```
": internal error: 010101_(5000 + 54)
icc: error #10014: problem during multi-file optimization compilation (code 4)
```
Maybe you can check if its working for you without `cython` installed?

Besides, `setup.py` still relies on an installed `numpy`.

anntzer commented on sksparse/cholmod.pyx:
```diff
@@ -31,8 +31,6 @@
 
 #cython: binding = True
 #cython: language_level = 3
-#distutils: include_dirs = NP_GET_INCLUDE /usr/include/suitesparse
```
Debian installs the headers to /usr/include/suitesparse (but Arch Linux to /usr/include) so you need to keep the former option too.

jor- replied:
I readded /usr/include/suitesparse.

anntzer replied:
I would completely strip out the distutils section here... the entry in `setup.py` is enough and having conflicting info is the last thing we want.

jor- replied:
That is true. I removed the distutils section in cholmod.pyx.

anntzer commented on setup.py:
```diff
@@ -50,9 +49,11 @@
+          # library_dirs and include_dirs keywords in the lines below.
+          ext_modules = [Extension("sksparse.cholmod", ["sksparse/cholmod.pyx"],
+              include_dirs=[np.get_include()],
+              library_dirs=[],
```
If library_dirs is not needed, just drop it?

anntzer commented:
You probably also want to update travis.yml and environment.yml to not install cython explicitly anymore.

jor- commented:
I updated travis.yml so that it is tested once without `cython` preinstalled and once with `cython` preinstalled . I also removed `cython` from environment.yml.

jor- commented:
It is working now as expected.

anntzer commented on .travis.yml:
```diff
@@ -4,7 +4,12 @@ addons:
-    - pip install cython scipy
+    - pip install 'setuptools>=18'
+    - pip install $PRE_INSTALL
+    - pip list
```
Remove env and replace these three lines by `pip install setuptools>=18 scipy`?

jor- replied:
I merged the two lines 
```
pip install 'setuptools>=18'
pip install $PRE_INSTALL
```
and removed
```
pip list
```

But I have to use env, otherwise I will not have the tests once with `cython` installed and once without.

anntzer commented on .travis.yml:
```diff
@@ -4,7 +4,10 @@ addons:
         packages:
             - libsuitesparse-dev
+env:
+  - PRE_INSTALL='setuptools>=18 scipy cython'
```
I think you can remove *this* line, I wouldn't worry too much about it (if it works with no cython...).

jor- replied:
Okay, I removed the test with `cython` preinstalled.

anntzer commented:
Squash it and looks good to me.

jor- commented:
I squashed it.
